src/converse-register.js

```diff
@@ -241,6 +245,25 @@
                     return false;
                 },
 
+                chooseProvider: function (ev) {
+                    /* Method called when registration_domain is provided
+                     * which loads the registration form directly
+                     */
+                    var $form = $('form#converse-register');
+                            $form.find('input[name=domain]').hide();
+                            $form.find('input[type=submit]').hide()
+                                .after(tpl_registration_request({
+                                    cancel: _converse.__('Cancel'),
+                                    info_message: _converse.__('Requesting a registration form from the XMPP server')
+                                }));
+                            $form.find('button.button-cancel').hide();
```

Comment:
Instead of manually hiding the cancel button here, you can hide it in the template itself with an `if` statement. You pass in another boolean value to `tpl_registration_request` and then check that value in the template to decide whether the cancel button should be shown or not.